collector/src/main/java/com/bitmovin/analytics/BitmovinAnalyticsConfig.java

```diff
@@ -300,4 +300,8 @@ public Context getContext() {
     public CollectorConfig getConfig() {
         return config;
     }
+
+    public void setConfig(CollectorConfig config) {
```

Comment:
Do we really want to allow setting of this? Because the issue is that once we allow setting a new `CollectorConfig` we are changing the reference.

So any code that stores the collectorConfig in a local variable (or field) will start doing weird stuff if someone changes the config with `setConfig`:

```
val config = analyticsConfig.getCollectorConfig();
config.getIsLive();
```

Example in Kotlin:
```
data class Config(var s: String)
class Test(var config: Config)

class Tests {
    @Test
    fun testSomething() {
        var x = Test(config = Config("foo"))
        val c = x.config
        x.config.s = "foo"
        assertThat(x.config.s).isEqualTo(c.s) // this is fine

        x.config = Config("bla")
        assertThat(x.config.s).isEqualTo(c.s) // this fails as the c.s is no longer pointing to the same Config as x.config
    }
}
```